packages/core/src/controllers/NetworkController.ts

```diff
@@ -190,25 +191,20 @@ export const NetworkController = {
   },
 
   async switchActiveNetwork(network: NetworkControllerState['caipNetwork']) {
-    const networkControllerClient = ChainController.getNetworkControllerClient()
-    await networkControllerClient.switchCaipNetwork(network)
+    const sameChain = network?.chain === ChainController.state.activeChain
 
-    const chain = ChainController.state.multiChainEnabled
-      ? network?.chain
-      : ChainController.state.activeChain
+    let networkControllerClient: NetworkControllerState['_client'] = undefined
 
-    if (!chain) {
-      throw new Error('chain is required to switch active network')
-    }
-
-    if (!network) {
-      throw new Error('network is required to switch active network')
+    if (sameChain) {
```

Comment:
Why do we need a special case for same chain?

Reply:
There is a case in multi-chain flow; 
- You're in Solana network (`solana` chain active on ChainController, thus you'll be using the `networkControllerClient` of the Solana adapter).
- You want to switch to Polygon, but the `evm` chain's active network is `Ethereum`.

In this case, you need to use the `evm`s `networkControllerClient` and call like the following;
```js
await networkControllerClient?.switchCaipNetwork(network)
```

Reply:
Well, actually we can get rid of the `sameChain` control 🤔 Yeah totally, just get the network's chain's `networkControllerClient` and proceed:

```js
let networkControllerClient: NetworkControllerState['_client'] = network
  ? ChainController.state.chains.get(network.chain)?.networkControllerClient
  : undefined

if (!networkControllerClient) {
  throw new Error('networkControllerClient not found for given network object')
}

...
```

Reply:
exactly this 💪 💯